### ISSUE TITLE
Add subdomain with plain HTTP listening on HTTPS port

### ISSUE THREAD

**Issue by grzegorz-dubicki** [lgarron/badssl.com#141]
I would find it useful to have a subdomain named, for example, not-https.badssl.com, where plain HTTP server would listed on HTTPS port.

(Note that it would be different that http.badssl.com, where a proper HTTPS server listens but redirects to HTTP URL on the same host.)

**Comment by lgarron** [comment]
Hmm, I believe serving HTTP and HTTPS on port 443 on the same server is possible but not straightforward.

Is there a particular use case you have in mind?

**Comment by grzegorz-dubicki** [comment]
I have encountered a plain HTTP servers on 443 port in the past at least once, but sadly I don't remember the exact case.

Anyway I thought that it would be nice to have a tool to test and detect that.

But now that you mentioned it: I also don't think that "serving HTTP and HTTPS on port 443 on the same server" is possible at all unless you have 2 IP addresses. Sorry, I haven't actually thought about the implementation when I was creating this issue..

**Comment by lgarron** [comment]
> I have encountered a plain HTTP servers on 443 port in the past at least once, but sadly I don't remember the exact case.

Yeah, it happens once in a while. I don't expect major browsers to do anything funky for it, but it would be interesting if e.g. they suppress the port in UI even if the scheme is HTTP and the port is 443.

> serving HTTP and HTTPS on port 443 on the same server" is possible at all unless you have 2 IP addresses

It's possible. Example: https://davidben.net:443
But not straightforward.
